lib/routes/gamer520/index.ts

```diff
@@ -0,0 +1,81 @@
+import { Data, DataItem, Route } from '@/types';
+import got from '@/utils/got';
+import { parseDate } from '@/utils/parse-date';
+import { load } from 'cheerio';
+import { Context } from 'hono';
+
+export const route: Route = {
+    path: '/:category?/:order?/:price?',
+    categories: ['game'],
+    example: '/gamer520/switchyouxi',
+    parameters: {
+        category: '分类，见下表',
+        order: '排序，见下表',
+        price: '价格, 全部:0;免费1;付费:2;',
+    },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    name: '全球游戏交流中心',
+    maintainers: ['xzzpig'],
+    handler,
+    url: 'www.gamer520.com/',
+    description: `分类
+
+  | Switch游戏下载 | 金手指 | 3A巨作 | switch主题 | PC游戏 | PC游戏下载 | 模拟器大全 | 听我所听 |
+  | --- | --- | --- | --- | --- | --- | --- | --- |
+  | switchyouxi | jinshouzhi | 3ajuzuo | zhuti | pcgame | pcgame | zhangji | ting |
+  
+  排序
+  
+  | 发布日期 | 修改日期 | 评论数量 | 随机 | 热度 |
+  | --- | --- | --- | --- | --- |
+  | date | modified | comment_count | rand | hot |`,
+};
+
+async function handler(ctx?: Context): Promise<Data> {
+    const category = ctx?.req.param('category') ?? 'switchyouxi';
+    const order = ctx?.req.param('order');
+    const price = ctx?.req.param('price');
+
+    const currentUrl = new URL(`https://www.gamer520.com/${category}`);
+    if (order !== undefined) {
+        currentUrl.searchParams.set('order', order);
+    }
+    if (price !== undefined) {
+        currentUrl.searchParams.set('cao_type', price);
+    }
+
+    const response = await got(currentUrl);
+    const $ = load(response.data as any);
+
+    const selector = `main article`;
+    const items: DataItem[] = $(selector)
+        .toArray()
+        .map((item) => {
+            const $item = $(item);
+            const $titleItem = $item.find('.entry-title a');
+            return {
+                title: $titleItem.text().trim(),
+                link: $titleItem.attr('href')!,
+                pubDate: parseDate($item.find('time').attr('datetime')?.trim() ?? ''),
+                category: $item
+                    .find('a[rel="category"]')
+                    .toArray()
+                    .map((a) => $(a).text().trim()),
+                description: $item.html() ?? '',
+            };
+        });
```

Comment:
Please use [WordPress REST API](https://developer.wordpress.org/rest-api/) instead.
The API endpoint references are listed at `/wp-json/wp/v2`.

You can list recent posts using `/wp-json/wp/v2/posts`. If you want categories, tags and author for each post, you can add the `_embed` search parameter when calling the posts API.

To adjust the no. of articles returned in the posts API, you can pass the `per_page` search parameter.

---
For categories, you can find the category id first using `/wp-json/wp/v2/categories` with the search parameter `slug`. The id for `switchyouxi` is `2`. Then you can list recent posts with the required tag using /wp-json/wp/v2/posts with the search parameter `categories`, which value will be `2` for `switchyouxi`.

Reply:
wow, got it. I'll revise it.

Reply:
A sad story, this site is more anti-bot than I thought. But the strange thing is that when I use the `curl` request, the request succeeds even if the UA is not set. But when I use `got` to request, even if I set up UA, the request will be recognized and blocke ip immediately. I don't know much about `got`, is there any characteristic of `got` that can be accurately recognized? (I tried setting ua and http2 parameters, but it didn't work)
Otherwise, I think `Puppeteer` would be the final solution